src/runtime/metal.cpp

```diff
@@ -426,9 +426,26 @@ WEAK command_buffer_completed_handler_block_descriptor_1 command_buffer_complete
     0, sizeof(command_buffer_completed_handler_block_literal)};
 
 WEAK void command_buffer_completed_handler_invoke(command_buffer_completed_handler_block_literal *block, mtl_command_buffer *buffer) {
```

Comment:
Is there no way to pass a valid user_context to this callback?

Reply:
I can enable that, sure.  But I'm not quite clear what should happen inside the callback then, with the `user_context`.  Is the idea users would override this with their own implementation and do stuff with the `user_context`?

Reply:
Yes. Code in google does this regularly... though the print and error handlers don't really use user_context, it's good form to pass in a valid value at all times unless there's a compelling reason not to.

Reply:
Ok, let me remember how to stash a `void*` pointer in the right place.  I think users would likely override with an Objective C implementation, but makes sense that we should pass it through.